packages/multi-test/src/app.rs

```diff
@@ -903,4 +911,172 @@ mod test {
             .unwrap();
         assert_eq!(state.beneficiary, random);
     }
+
+    mod replay_data_overwrite {
+        use cosmwasm_std::to_binary;
+
+        use crate::test_helpers::EmptyMsg;
+
+        use super::*;
+
+        fn make_echo_submsg(
+            contract: Addr,
+            data: impl Into<Option<&'static str>>,
+            sub_msg: Vec<SubMsg<Binary>>,
+        ) -> SubMsg<Binary> {
```

Comment:
Why do you need `SubMsg<Binary>`?

They type `T` here just controls the type of CosmosMsg::Custom variant, so we can compile with code that uses it. We return an error if anyone actually returns those variants in multi-test (this is a future TODO to support it)